app/lib/ingestors/google/gmail.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Ingestors::Google::Gmail
+  def initialize(user)
+    client = Signet::OAuth2::Client.new(
+      token_credential_uri: 'https://oauth2.googleapis.com/token',
+      client_id: ENV.fetch('GOOGLE_CLIENT_ID', nil),
+      client_secret: ENV.fetch('GOOGLE_CLIENT_SECRET', nil),
+      refresh_token: user.google_oauth_credentials['refresh_token']
+    )
+    client.fetch_access_token!
+
+    @client = client
+    @user = user
+  end
+
+  def ingest
+    messages = gmail_client.list_user_messages('me', q: 'is:unread')&.messages
```

Comment:
Going to change this query in the next PR. Not sure atm what all messages do we need to fetch